diff -Nur sms/bramka.cc sms.new/bramka.cc
--- sms/bramka.cc	Mon Dec  2 09:44:18 2002
+++ sms.new/bramka.cc	Tue Jun 10 07:47:28 2003
@@ -5,7 +5,7 @@
  * 29.11.2002, Piotr Godowski, <flash@v-lo.krakow.pl>
  */
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include "request.h"
 #include "headers.h"
 #include "sms.h"
diff -Nur sms/czat.cc sms.new/czat.cc
--- sms/czat.cc	Mon Dec  2 09:44:18 2002
+++ sms.new/czat.cc	Tue Jun 10 07:47:28 2003
@@ -3,7 +3,7 @@
  * Wysyla sms'a poprzez bramke Idea.czat
  */
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include "request.h"
 #include "headers.h"
 #include "sms.h"
diff -Nur sms/debug.cc sms.new/debug.cc
--- sms/debug.cc	Sat Dec 28 17:35:42 2002
+++ sms.new/debug.cc	Tue Jun 10 07:47:28 2003
@@ -20,7 +20,7 @@
 void Debug::Open()
 {
 #ifdef DEBUG
-     if (!str) str = new ofstream(name);
+     if (!str) str = new std::ofstream(name);
 #endif
 }
 
@@ -31,7 +31,7 @@
     
     char buf[MAX_LOG_LINE_LEN];
     va_list args;
-    ofstream &stream = *str;
+    std::ofstream &stream = *str;
 
     va_start(args, msg);
     VSPRINTF(buf, sizeof(buf)-1, msg, args);
diff -Nur sms/debug.h sms.new/debug.h
--- sms/debug.h	Sun Jun  1 23:28:31 2003
+++ sms.new/debug.h	Tue Jun 10 07:47:28 2003
@@ -1,7 +1,7 @@
 #ifndef __debug_h
 #define __debug_h
 
-#include <fstream.h>
+#include <fstream>
 
 #define LOG_FILE "sms.log"
 #define MAX_LOG_LINE_LEN 32768
@@ -11,7 +11,7 @@
 
 class Debug
 {
-      ofstream *str;
+      std::ofstream *str;
       char *name;
 
       void Open();
diff -Nur sms/era.cc sms.new/era.cc
--- sms/era.cc	Mon Jun  2 00:12:52 2003
+++ sms.new/era.cc	Tue Jun 10 07:47:28 2003
@@ -1,5 +1,5 @@
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include "request.h"
 #include "headers.h"
 #include "sms.h"
diff -Nur sms/freesms.cc sms.new/freesms.cc
--- sms/freesms.cc	Mon Dec  2 09:44:18 2002
+++ sms.new/freesms.cc	Tue Jun 10 07:47:28 2003
@@ -6,7 +6,7 @@
 *****************************************************************/
 
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include "request.h"
 #include "headers.h"
 #include "sms.h"
diff -Nur sms/hulp.cc sms.new/hulp.cc
--- sms/hulp.cc	Mon Jun  2 00:12:33 2003
+++ sms.new/hulp.cc	Tue Jun 10 07:47:28 2003
@@ -1,5 +1,5 @@
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include "request.h"
 #include "headers.h"
 #include "sms.h"
diff -Nur sms/main.cc sms.new/main.cc
--- sms/main.cc	Mon Jun  2 00:18:39 2003
+++ sms.new/main.cc	Tue Jun 10 07:47:28 2003
@@ -9,8 +9,8 @@
 #include <signal.h>
 #include <unistd.h>
 #include <ctype.h>
-#include <iostream.h>
-#include <strstream.h>
+#include <iostream>
+#include <sstream>
 #include <string>
 #include "vsprintf.h"
 #include "headers.h"
@@ -290,18 +290,12 @@
 
 	if (start_arg >= argc) {
 		// pobieramy ze stdin
-		strstream s;
-		char *n;
-		int len;
+		std::stringstream s;
+		string msg;
 
 		cin >> s.rdbuf();
-		len = s.pcount();
-		n = s.str();
-		strncpy(args.sms, (n) ? n : "", sizeof(args.sms)-1);
-		if (len < sizeof(args.sms)) {
-			args.sms[len] = 0;
-		}
-		s.freeze(0);
+		msg = s.str();
+		strncpy(args.sms, msg.c_str(), sizeof(args.sms)-1);
 	} else {
 		// pobieramy z linii polecen
 		string msg;
diff -Nur sms/pf.cc sms.new/pf.cc
--- sms/pf.cc	Mon Dec  2 09:44:45 2002
+++ sms.new/pf.cc	Tue Jun 10 07:47:28 2003
@@ -10,7 +10,7 @@
 */
 
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include "request.h"
 #include "headers.h"
 #include "sms.h"
diff -Nur sms/request.cc sms.new/request.cc
--- sms/request.cc	Mon Jun  2 00:17:35 2003
+++ sms.new/request.cc	Tue Jun 10 07:47:28 2003
@@ -2,8 +2,8 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <strstream.h>
-#include <iostream.h>
+#include <sstream>
+#include <iostream>
 #include "socket.h"
 #include "request.h"
 #include "sms.h"
@@ -171,7 +171,6 @@
     delete buf;
     return ret_buf; 
 }
-
 void HTTP_Request::ParseHeaders(Headers &hd, const char *buf, int buf_len)
 {
     // pierwsza linijka zawiera tekst: HTTP/1.1 200 OK
@@ -179,7 +178,7 @@
     // konczone sa dodatkowo znakiem \r (powrot karetki)
 
     char *line = new char[MAX_LINE_LEN], *c;
-    istrstream is(buf, buf_len);
+    std::istringstream is(buf);
     
     // najpierw linia ze statusem operacji
     is.getline(line, MAX_LINE_LEN);
diff -Nur sms/smsc.cc sms.new/smsc.cc
--- sms/smsc.cc	Sat Dec 28 18:59:17 2002
+++ sms.new/smsc.cc	Tue Jun 10 07:47:28 2003
@@ -5,7 +5,7 @@
  * 17.12.2002, Piotr Godowski, <flash@v-lo.krakow.pl>
  */
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include "request.h"
 #include "headers.h"
 #include "sms.h"
diff -Nur sms/supersms.cc sms.new/supersms.cc
--- sms/supersms.cc	Sat Dec 28 18:26:45 2002
+++ sms.new/supersms.cc	Tue Jun 10 07:47:28 2003
@@ -7,7 +7,7 @@
 *****************************************************************/
 
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <ctype.h>
 #include "request.h"
